components/form/pta/src/index.js

```diff
@@ -5,19 +5,37 @@ import {paramsToQueryString} from './querystring'
 const BASE_CLASS = 'sui-FormPta'
 const CONTENT_CLASS = `${BASE_CLASS}-content`
 const REMOVE_DRAFT = 'REMOVE_DRAFT'
+const SUBMIT_SUCCESS_EVENT_TYPE = 'SUBMIT_FORM_SUCCEEDED'
+const SUBMIT_ERROR_EVENT_TYPE = 'SUBMIT_FORM_FAILED'
+const MESSAGE_EVENT_TYPE = 'message'
 
 class FormPta extends Component {
   constructor(props) {
     super(props)
 
-    const {formUrl: BASE_URL, ...settings} = this.props
+    const {formUrl: BASE_URL, onSubmit, onError, ...settings} = this.props
     const QUERY = paramsToQueryString(settings)
     const formUrl = `${BASE_URL}?${QUERY}`
-
+    this.handleMessage = this.handleMessage.bind(this)
     this.state = {
       formUrl
     }
   }
+
+  componentWillMount() {
+    window.addEventListener(MESSAGE_EVENT_TYPE, this.handleMessage)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener(MESSAGE_EVENT_TYPE, this.handleMessage)
+  }
+
+  handleMessage({data: {type}}) {
+    const {onSubmit, onError} = this.props
```

Comment:
You might do these `required` props or having a defaultValue to avoid possible problems.

Reply:
These are optional function callback fields. Which default value should i set?

Reply:
Default value then should be an empty function in order your code to not break if the user of the component doesn't pass a function as a prop.